src/transformers/pipelines/image_text_to_text.py

```diff
@@ -0,0 +1,466 @@
+                containing the text of the message and the type of the message. The type of the message can be either
+                'text' or 'image'. If the type is 'image', no text is needed.
+            return_tensors (`bool`, *optional*, defaults to `False`):
+                Returns the tensors of predictions (as token indices) in the outputs. If set to
+                `True`, the decoded text is not returned.
+            return_text (`bool`, *optional*):
+                Returns the decoded texts in the outputs.
+            return_full_text (`bool`, *optional*, defaults to `True`):
+                If set to `False` only added text is returned, otherwise the full text is returned. Cannot be
+                specified at the same time as `return_text`.
+            continue_final_message( `bool`, *optional*): This indicates that you want the model to continue the
+                last message in the input chat rather than starting a new one, allowing you to "prefill" its response.
+                By default this is `True` when the final message in the input chat has the `assistant` role and
+                `False` otherwise, but you can manually override that behaviour by setting this flag.
+
+        Return:
+            A list or a list of list of `dict`: Each result comes as a dictionary with the following key (cannot return a combination
+            of both `generated_text` and `generated_token_ids`):
+
+            - **generated_text** (`str`, present when `return_text=True`) -- The generated text.
+            - **generated_token_ids** (`torch.Tensor`, present when `return_tensors=True`) -- The token
+                ids of the generated text.
+            - **input_text** (`str`) -- The input text.
+        """
+        batch_size = kwargs.get("batch_size", 1)
+
+        if images is None and text is None:
+            raise ValueError("You must at least provide either text or images.")
+
+        if isinstance(text, (list, tuple, KeyDataset) if is_torch_available() else (list, tuple)) and isinstance(
+            text[0], (list, tuple, dict)
+        ):
+            # We have one or more prompts in list-of-dicts format, so this is chat mode
+
+            if isinstance(text[0], dict):
+                return super().__call__(Chat(text, images), **kwargs)
+            else:
+                if images is None:
+                    images = [None] * len(text)
+                chats = [Chat(chat, image) for chat, image in zip(text, images)]  # 🐈 🐈 🐈
+                return super().__call__(chats, **kwargs)
+
+        # If we are not in chat mode, we need both images and text
+        if images is None or text is None:
+            raise ValueError("You must provide both images and text when not using chat templates.")
+
+        if not isinstance(images, (list, tuple)):
+            images = [images]
+        if isinstance(text, str):
+            text = [text] * len(images)
```

Comment:
Maybe there is something to change here indeed in the case where text is a string. 
Right now we assume that if text is a string and there is multiple images (non nested), the prompt should be applied to each image independently.
It would maybe make more sense to consider as a multiimage prompt.
But yes in that specific case we need to explicitly pass in a nested list of image `[[image1, image2,...]]` with `text = "prompt"` , which is weird.
I will try changing that and see if I run into any problems.

Reply:
After testing a bit, it looks like most models which work without image tokens also don't accept multiple images for one prompt (kosmos 2, blip/blip2, paligemma, udop, GIT), and the errors are not always caught, so it's not clear to the user why the pipeline crashes.
The one I found that does work is llava-onevision, but then the output seems to be garbage, so I'm not sure we should encouraged this behavior. Wdyt?
On that last point, it looks like a lot of models which support chat templates do not work well with "raw" text inputs. Should we raise a warning when a model supports chat template, but the input is raw text?

Reply:
Right now I settled on this:
- if there are image tokens, we accept one prompt for multiple images, no matter if the images are nested or not.
- if there are no image tokens and one prompt for multiple images, the prompt will be repeated for each image, and raise a warning.

Reply:
oke, agreed, makes sense! Thanks